recipe/meta.yaml

```
@@ -13,18 +12,19 @@ source:
 build:
   entry_points:
     - instructor = instructor.cli.cli:app
-  noarch: python
   script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  skip: true  # [py<310]
   number: 0
-
+# Though it is available upstream for 3.9, we’ll limit the package to Python 3.10 and above,
+# since the newer typing syntax breaks all 3.9 builds.  
 requirements:
   host:
-    - python {{ python_min }}
+    - python
     - poetry-core
     - hatchling
     - pip
   run:
-    - python >={{ python_min }}
+    - python
     - openai >=1.52.0,<2.0.0
     - pydantic >=2.8.0,<3.0.0
     - docstring_parser >=0.16.0,<0.17.0
```

```suggestion
    - docstring_parser >=0.16.0,<1.0
```
https://github.com/instructor-ai/instructor/blob/1.7.9/pyproject.toml#L8

There are a few others that are off as well such as `typer`, `requests`, `jiter` and `jinja2`.

Most of the `eun-constrained` applies only to the `instructor[test-docs]` target. The ones that don't need updating.

---

Unfortunately, there is a error regarding the python 3.9 ability to successfully build.  Though it is available upstream for 3.9, I've limited the package to Python ≥3.10 because the upstream “support” for 3.9 is causing errors:
> * TypeError: Unable to evaluate type annotation 'str | Path' (the new | syntax just won’t parse in conda-> * > * build or Prefect).

Conda-forge already pulled the plug on 3.9 for the same reason i'm assuming, so I figured following their lead to keep our CI from falling over. If someone really needs 3.9, we'd have to figure out a workaround

---

I linked the workaround in my comment. There's a package that backports 3.10's typing changes, if we end up needing 3.9 for something.

---

Are you referring to `docstring-parser` vs `docstring_parser`? Sorry, that's what I see regarding the link. We currently have `docstring_parser` available that I've just built, versus` docstring-parser `which I think may be the same. Do I need to build another package utilizing `docstring- parser`?

---

> They have test cases that could be run. We could also build for 3.9 if we wanted it. [The support package](https://github.com/AnacondaRecipes/eval-type-backport-feedstock/blob/main/recipe/meta.yaml) looks pretty simple to build.

This one goes to docstring parser? Should be `eval-type-backport`. In any case, not essential. Just wanted it referenced in case we later decide we do need 3.9 support.